spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -78,88 +79,102 @@ public void setDefaultTopic(String defaultTopic) {
-	public Future<RecordMetadata>  convertAndSend(String topic, K key, V data) {
+	public Future<RecordMetadata>  send(String topic, K key, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<>(topic, key, data);
 		return doSend(producerRecord);
 	}
 
 	@Override
-	public Future<RecordMetadata>  convertAndSend(String topic, int partition, K key, V data) {
+	public Future<RecordMetadata> send(String topic, int partition, V data) {
+		ProducerRecord<K, V> producerRecord = new ProducerRecord<K, V>(topic, partition, null, data);
+		return doSend(producerRecord);
+	}
+
+	@Override
+	public Future<RecordMetadata>  send(String topic, int partition, K key, V data) {
 		ProducerRecord<K, V> producerRecord = new ProducerRecord<>(topic, partition, key, data);
 		return doSend(producerRecord);
 	}
 
 
 	@Override
-	public RecordMetadata syncConvertAndSend(V data) throws InterruptedException, ExecutionException {
-		Future<RecordMetadata> future = convertAndSend(data);
+	public RecordMetadata syncSend(V data) throws InterruptedException, ExecutionException {
+		Future<RecordMetadata> future = send(data);
 		flush();
 		return future.get();
 	}
 
 	@Override
-	public RecordMetadata syncConvertAndSend(K key, V data) throws InterruptedException, ExecutionException {
-		Future<RecordMetadata> future = convertAndSend(key, data);
+	public RecordMetadata syncSend(K key, V data) throws InterruptedException, ExecutionException {
+		Future<RecordMetadata> future = send(key, data);
 		flush();
 		return future.get();
 	}
 
 	@Override
-	public RecordMetadata syncConvertAndSend(int partition, K key, V data)
+	public RecordMetadata syncSend(int partition, K key, V data)
 			throws InterruptedException, ExecutionException {
-		Future<RecordMetadata> future = convertAndSend(partition, key, data);
+		Future<RecordMetadata> future = send(partition, key, data);
 		flush();
 		return future.get();
 	}
 
 	@Override
-	public RecordMetadata syncConvertAndSend(String topic, V data) throws InterruptedException, ExecutionException {
-		Future<RecordMetadata> future = convertAndSend(topic, data);
+	public RecordMetadata syncSend(String topic, V data) throws InterruptedException, ExecutionException {
+		Future<RecordMetadata> future = send(topic, data);
 		flush();
 		return future.get();
 	}
 
 	@Override
-	public RecordMetadata syncConvertAndSend(String topic, K key, V data)
+	public RecordMetadata syncSend(String topic, K key, V data)
 			throws InterruptedException, ExecutionException {
-		Future<RecordMetadata> future = convertAndSend(topic, key, data);
+		Future<RecordMetadata> future = send(topic, key, data);
 		flush();
 		return future.get();
 	}
 
 	@Override
-	public RecordMetadata syncConvertAndSend(String topic, int partition, K key, V data)
+	public RecordMetadata syncSend(String topic, int partition, V data)
+			throws InterruptedException, ExecutionException {
+		Future<RecordMetadata> future = send(topic, partition, data);
+		return future.get();
```

Comment:
Yeah, I was thinking it would be a convenience to save the user from 2 statements.

But I suppose he could use

template.send(...).get()

I'll add another commit.